lib/src/config/address.rs

```diff
@@ -0,0 +1,63 @@
+use std::io;
+use std::fmt;
+use std::str::FromStr;
+use std::convert::TryFrom;
+use std::path::PathBuf;
+use std::net::{SocketAddr, ToSocketAddrs};
+
+/// An enum corresponding to the address to serve on
+#[derive(Debug, Clone, PartialEq)]
+pub enum Address {
+    /// The TCP address
+    Tcp(SocketAddr),
+    /// Stores the validated path to the unix domain socket.
+    Unix(PathBuf),
+}
+
+impl FromStr for Address {
+    type Err = io::Error;
+
+    fn from_str(s: &str) -> ::std::result::Result<Self, Self::Err> {
+        if s.starts_with("unix://") {
+            let path_str = &s[7..];
+            let path: PathBuf = path_str.into();
+            Ok(Address::Unix(path))
```

Comment:
This can be one line: `Address::Unix(s[7..].into())`. Better to make these prefixes `const`s though. Something like:

```rust
const UNIX_PREFIX: &str = "unix://";
const TCP_PREFIX: &str = "tcp://";

if string.starts_with(UNIX_PREFIX) {
    Ok(Address::Unix(string[UNIX_PREFIX.len()..].into()))
} else string.starts_with(TCP_PREFIX) {
    ...
} else {
    ...
}
```